spark/src/main/java/io/ddf/spark/content/SchemaHandler.java

```diff
@@ -106,5 +127,260 @@ public static Schema getSchemaFromDataFrame(DataFrame rdd) {
     List<Object> listValues = FactorIndexer.getFactorMapForColumn(this.getDDF(), columnName).values();
     return listValues;
   }
+
+  /**
+   * applySchema(Schema newSchema) will do apply change column type from current to new data type. This function return
+   * new DDF and also statistic when its applied new schema
+   *
+   * @param schema will use to apply for current DDF
+   * @return Tuple2<SparkDDF,ApplySchemaStatistic> +SparkDDF: new sparkddf which applied new schema already
+   * +ApplySchemaStatistic:  applied schema statistic @see ApplySchemaStatistic
+   */
+  public Tuple2<SparkDDF, ApplySchemaStatistic> applySchema(Schema schema) throws DDFException {
+    long startApplySchema = System.currentTimeMillis();
+    // Keep info to Convert column at <pos> to <ColumnType>
+    List<Tuple3<Integer, String, Schema.ColumnType>> listConvertColumns = getListConvertColumn(this.getSchema(), schema);
+
+    try {
+      SparkDDFManager manager = (SparkDDFManager) this.getManager();
+//      SparkContext sparkContext = manager.getSparkContext();
+      Accumulator<ApplySchemaStatistic> statisticAccumulator = manager.getSparkContext().accumulator(new ApplySchemaStatistic(), ApplySchemaStatisticParam.getInstance());
+
+      JavaRDD<Row> rdd = ((SparkDDF) this.getDDF()).getRDD(Row.class).toJavaRDD();
+      JavaRDD<Row> filteredRdd = rdd.filter(row -> convertRowType(row, listConvertColumns, statisticAccumulator));
```

Comment:
do you mean this code @Huandao0812 :` df.select(df("colA").cast("int"))`
So how do we get statistic for sample data & how many row failed on this column ?
btw, what is the reason to make this code slow? I think it's also O(1) ?